packages/ra-core/src/controller/usePrevNextController.ts

```diff
@@ -124,37 +125,78 @@ export const usePrevNextController = <RecordType extends RaRecord = any>(
         );
     }
 
-    const [storedParams] = useStore<StoredParams>(
+    const [storedParams] = useStore<Partial<ListParams>>(
         storeKey || `${resource}.listParams`,
         {
-            filter: {},
-            order: sort.order,
-            sort: sort.field,
+            filter: filterDefaultValues,
+            order: initialSort.order,
+            sort: initialSort.field,
+            page: 1,
+            perPage: 10,
         }
     );
 
-    const { data, error, isLoading } = useGetList<RecordType>(
+    const dataProvider = useDataProvider();
+    const queryClient = useQueryClient();
+    const pagination = { page: 1, perPage: limit };
+    const sort = {
+        field: storedParams.sort,
+        order: storedParams.order,
+    };
+    const filter = { ...storedParams.filter, ...permanentFilter };
+    const { meta, ...otherQueryOptions } = queryOptions;
+    const params = { pagination, sort, filter, meta };
+
+    // try to use data from the cache first
+    const queryData = queryClient.getQueryData<{
+        data: RaRecord[];
+        total: number;
+    }>([
         resource,
+        'getList',
         {
-            sort: {
-                field: storedParams.sort,
-                order: storedParams.order,
+            ...params,
+            pagination: {
+                page: storedParams.page,
+                perPage: storedParams.perPage,
```

Comment:
Maybe you shouldn't include `meta` here, while interrogating the cache, because `meta` from `<PrevNextButton>` may not be the same as `meta` from `<List>`.

We may need to provide a different `meta` prop for interrogating the cache and for actually calling `getList` in the context of this controller.

Reply:
I can't determine if it's better to include it or not. Let's wait and see if the problem arises before making the API more complex.